src/main/kotlin/com/softwareangels/storeapp/batch/ProductsJob.kt

```diff
@@ -0,0 +1,119 @@
+package com.softwareangels.storeapp.batch
+
+import com.softwareangels.storeapp.dto.ProductDTO
+import com.softwareangels.storeapp.model.Product
+import com.softwareangels.storeapp.model.Store
+import com.softwareangels.storeapp.repository.ProductRepository
+import com.softwareangels.storeapp.repository.StoreRepository
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.ResponseEntity
+import org.springframework.retry.annotation.Backoff
+import org.springframework.retry.annotation.Recover
+import org.springframework.retry.annotation.Retryable
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Service
+
+@Service
+class ProductsJob(
+    val allStoreDataRepository: ProductRepository,
+    val storeRepository: StoreRepository) : BaseJob() {
+
+    val log = LoggerFactory.getLogger(ProductsJob::class.java)
+
+    @Value("\${app.api.products}")
+    val apiHost: String = ""
+
+    var page: Int = 0
+
+    val sizes = arrayOf("XS", "S", "L", "XL", "XXL", "XXXL")
+
+
+    @Scheduled(fixedDelayString = "\${app.scheduler.delay}", initialDelayString = "\${app.scheduler.initial-delay.products}")
+    @Retryable(value = [JobException::class], maxAttemptsExpression = "\${app.retry.max-attempts.products}",
+        backoff = Backoff(delayExpression = "\${app.retry.delay}"))
+    fun getProducts() {
+        log.info("Running ProductsJob")
+        do {
+            try {
+                val res = httpGet(apiHost, page, Array<ProductDTO>::class.java) as ResponseEntity<Array<ProductDTO>>
+
+                if(res.body?.isEmpty() == true) break
+
+                for (response in res.body!!) {
+                    val stores = storeRepository.findByName(response.store)
+                    val allStoreData = allStoreDataRepository.findByProductSeasonAndProductModel(response.season, response.product)
+
+                    if (allStoreData.isEmpty()) {
+                        generateProductData(stores!!, response)
+                    } else {
+                        updateProductData(allStoreData, stores!!, response)
+                    }
+                }
+
+                page += 1
+
+            } catch (ex: Exception) {
+                log.error("Error running ProductsJob on page $page. Error: ${ex.message}")
```

Comment:
Do you see any limitation with the way you are logging errors here?

Reply:
Yeah, this can be improve to handle the exception into the for loop, so we can track the specific data that throws the exception. 
Now, what we have is just the page, and the data from it is dynamic.

Reply:
Indeed.
What about the stack trace? What if the exception has no message? Why are you concatenating the exception message in a string instead of using `log.error(message, cause)`?